framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -305,6 +306,21 @@
     /** Mapping from an Element to the source Tree of the declaration. */
     private final Map<Element, Tree> elementToTreeCache;
 
+    /** Viewpoint adaptor to perform viewpoint adaptation */
+    protected final ViewpointAdaptor<?> viewpointAdaptor;
```

Comment:
Can you see how this related to the GenericAnnotatedTypeFactory#DependentTypesHelper that was added since your PR?
Should this also be in the GenericAnnotatedTypeFactory?
How do the two mechanisms relate? I remember we discussed this before, but please have another look at this.